<?xml version="1.0"?>
<launch>
  <arg name="bag" default="test.bag"/>
  <arg name="use_sim_time" default="true"/>

  <group>
    <!-- <push-ros-namespace namespace="utils"/> -->
    <node_container pkg="rclcpp_components" exec="component_container_mt" name="livox_util_container" namespace="" output="screen">
      <composable_node pkg="utils" plugin="utils::LivoxCloudMerger" name="livox_cloud_merger" namespace="">
        <param name="use_sim_time" value="$(var use_sim_time)"/>
        <param name="cloud_topic" value="/livox/lidar"/>
        <param name="pointcloud1_topic" value="/livox/lidar_192_168_1_101"/>
        <param name="pointcloud2_topic" value="/livox/lidar_192_168_1_102"/>
        <param name="output_topic" value="/livox/merged_cloud"/>
        <param name="frame_id" value="livox_frame"/>
        <param name="lidar_diffz" value="0.47"/>
        <param name="ring_num" value="10"/>
        <extra_arg name="use_intra_process_comms" value="true" />
      </composable_node>
    </node_container>
  </group>

  <!-- rosbag -->
  <executable cmd="ros2 bag play $(var bag)" output="screen" name="rosbag_play" />

  <!-- rviz -->
  <node pkg="rviz2" exec="rviz2" name="rviz2" output="screen" />
</launch>